xrpl/utils/txn_parser/get_balance_changes.py

```diff
@@ -0,0 +1,60 @@
+"""Parse balance changes of every account involved in the given transaction."""
+
+from decimal import Decimal
+from typing import List, Optional
+
+from xrpl.models import TransactionMetadata
+from xrpl.utils.txn_parser.utils import (
+    BalanceChanges,
+    get_quantities,
+    get_value,
+    group_by_account,
+    normalize_nodes,
+)
+from xrpl.utils.txn_parser.utils.nodes import NormalizedNode
+
+
+def get_balance_changes(metadata: TransactionMetadata) -> List[BalanceChanges]:
+    """
+    Parse all balance changes from a transaction's metadata.
+
+    Args:
+        metadata: Transactions metadata.
+
+    Returns:
+        All balance changes caused by a transaction.
+        The balance changes are grouped by the affected account addresses.
+    """
+    quantities = [
+        get_quantities(node, _compute_balance_change(node))
+        for node in normalize_nodes(metadata)
+    ]
+    return group_by_account(quantities)
+
+
+def _compute_balance_change(node: NormalizedNode) -> Optional[Decimal]:
+    """
+    Get the balance change from a node.
+
+    Args:
+        node: The affected node.
+
+    Returns:
+        The balance change.
+    """
+    value: Optional[Decimal] = None
+    new_fields = node.get("NewFields")
+    previous_fields = node.get("PreviousFields")
+    final_fields = node.get("FinalFields")
```

Comment:
If I'm understanding the flow of your code correctly, these don't need `.get` because you already used `.get` in `normalize_nodes`